crates/optimism/rpc/src/sequencer.rs

```diff
@@ -103,21 +152,58 @@ impl SequencerClient {
 struct SequencerClientInner {
     /// The endpoint of the sequencer
     sequencer_endpoint: String,
-    /// The HTTP client
-    http_client: Client,
+    /// The client
+    client: Client,
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
     use alloy_primitives::U64;
 
-    #[test]
-    fn test_body_str() {
-        let client = SequencerClient::new("http://localhost:8545");
+    #[tokio::test]
+    async fn test_http_body_str() {
+        let client = SequencerClient::new("http://localhost:8545").await.unwrap();
+
+        let request = client
+            .client()
+            .make_request("eth_getBlockByNumber", (U64::from(10),))
+            .serialize()
+            .unwrap()
+            .take_request();
+        let body = request.get();
+
+        assert_eq!(
+            body,
+            r#"{"method":"eth_getBlockByNumber","params":["0xa"],"id":0,"jsonrpc":"2.0"}"#
+        );
+
+        let condition = TransactionConditional::default();
+
+        let request = client
+            .client()
+            .make_request(
+                "eth_sendRawTransactionConditional",
+                (format!("0x{}", hex::encode("abcd")), condition),
+            )
+            .serialize()
+            .unwrap()
+            .take_request();
+        let body = request.get();
+
+        assert_eq!(
+            body,
+            r#"{"method":"eth_sendRawTransactionConditional","params":["0x61626364",{"knownAccounts":{}}],"id":1,"jsonrpc":"2.0"}"#
+        );
+    }
+
+    #[tokio::test]
+    #[ignore = "Start if WS is reachable at ws://localhost:8546"]
+    async fn test_ws_body_str() {
+        let client = SequencerClient::new("ws://localhost:8546").await.unwrap();
```

Comment:
This requires a client though. Can remove this test, but just in case, it works for me locally